### PR TITLE
Hipblaslt routing for prefill GEMMs

### PR DESCRIPTION
**Pull request by cagrikymk**
This PR updates GEMM routing so that hipblaslt is used for the fat GEMMs and Triton for skinny GEMMs for GPT-OSS.

### CODE REVIEW

File: vllm/model_executor/layers/utils.py [hunk 1]
```diff
@@ -110,29 +113,27 @@ def rocm_unquantized_gemm_impl(
     from vllm.platforms.rocm import on_gfx9
     k = weight.shape[1]
     m = weight.shape[0]
-    if x.is_contiguous():
```
Comment (dllehr-amd) [inline vllm/model_executor/layers/utils.py:113]: @cagrikymk  leave this contiguous check.  it will catch mllama.py issues with the non-contiguous tensor shape

Comment (cagrikymk) [reply vllm/model_executor/layers/utils.py:113]: @dllehr-amd Basically that logic is same as only ops.wvSplitK and ops.LLMM1 consume x_view. Since x might contain useful padding (which is the case for GPT-OSS), I moved contiguous related code block right before it is consumed. This shouldnt change how the code works. 

It is just to make sure torch.nn.functional.linear and gemm_a16w16 are called before the contiguous check as they only access x, not x_view.

**Review by dllehr-amd — APPROVED** [review]
Approved!
